spec.bs

```diff
@@ -837,6 +838,7 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
       |auctionConfig|, |global|, and |settings|.
     1. If |compWinner| is failure, return failure.
     1. If |compWinner| is not null:
+      1. If [=waiting until configuration input promises resolve=] given |auctionConfig| returns failure, return failure.

```

Comment:
Doing this inside a parallel queue will mean that all subsequent steps get delayed. That's probably fine in this case since the wait has the same condition for all of them, and so subsequent ones won't wait anymore, but that means you can probably move it up in front of the overall loop so that future reviewers don't worry about this.

Reply:
I am unclear as to what you mean by subsequent here given the, well, parallelism.

What this is trying to achieve, at any rate, is this:
1) Component auctions only need to wait for their own configuration to resolve to start bidding process.
2) Top-level auction needs the whole hierarchy before starting its scoring, but its own scoring happens after a component makes a bid (before this change, it could still have some promises hanging out).
We do not want to delay #1 for promises in top-level config, which is what I think your suggestion would do?

Actually, mistake #1: the top-level needs all component auctions to have a complete configuration, not just the one who bid its considering, since it needs to pass an entire AuctionConfig to scoreAd.

Mistake #2: we do wait for configs for nested auctions even if they did not actually make a bid in top-level auction, in order to type-check them (though that can race with other reasons for auction to fail).

... Working on fixing these...